app/Notification.jsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import './_ciq-notification.scss';
+
+import CloseIcon from './icons/ic-close.svg';
+import Warning from './icons/warning.svg';
+import Error from './icons/error.svg';
+import Success from './icons/success.svg';
+import Info from './icons/info.svg';
+
+
+const alertIconMaps = {
+    info: Info,
+    success: Success,
+    warning: Warning,
+    error: Error,
+};
+
+class Notification extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = { messages: [] };
+    }
+    onMessage(message, duration = 10) {
+        const messages = this.state.messages;
+        message.id = (new Date()).getTime();
+        message.type = message.type || 'warning';
+        message.hide = false;
+
+        messages.push(message);
+        this.setState({
+            messages,
+        });
+
+        if (duration > 0) {
+            setTimeout(() => this.onRemove(message.id), duration * 1000);
+        }
+    }
+    onRemove(id) {
+        let messages = this.state.messages.map((x) => {
+            x.hide = x.id === id ? true : x.hide;
+            return x;
+        });
+        this.setState({
+            messages,
+        });
+
+        setTimeout(() => {
+            messages = messages.filter(x => x.id !== id);
+            this.setState({
+                messages,
+            });
+        }, 300);
```

Comment:
@shayanbinary why we need the 300ms delay? what happens if OnMessage gets called with a new message within that 300ms?

Reply:
If `OnMessage` get called just new message appear, as you see we assign an id to each message `message.id = (new Date()).getTime();`.

Reply:
I see, makes sense now :)

Reply:
> message removing has an animation which animate 300ms so, first of all, I set hide=true to start animation then after the animated finish, I remove the message for the store

That's a good explanation 👍 . I think it should be in the code as well.